packages/flutter_tools/lib/src/build_system/targets/dart.dart

```diff
@@ -183,7 +184,9 @@ class KernelSnapshot extends Target {
   ];
 
   @override
-  List<Target> get dependencies => <Target>[];
+  List<Target> get dependencies => const <Target>[
+    GenerateLocalizationsTarget(),
```

Comment:
Have you considered moving the new `canSkip` check here? That is, not including it in the dependency list at all?

Reply:
This won't work for now because I store inputs/outputs at the target level. So if a target is totally excluded from the build, then we don't know enough to skip. I could change the serialization format so there is only a single file, but that would take more refactoring.

Reply:
Err, I mean that we don't know enough to clean up the old outputs

Reply:
You mean, like if someone removed the `l10n.yaml` file, you wouldn't be able to clean up the generated source file? Or something else?

Reply:
If I conditionally added or removed the entire target, I couldn't clean up the output because its tracked in a file controlled by that target.

With this change I can cleanup the output if that file is removed